src/bookshelf/src/bksw_wm_base.erl

```diff
@@ -15,28 +15,124 @@
 %% under the License.
 %%
 %% @author Ho-Sheng Hsiao <hosh@chef.io>
-%% @author Tim Dysinger <dysinger@chef.io>
+%% @author Tim Dysinger   <dysinger@chef.io>
+%% @author Lincoln Baker  <lbaker@chef.io>
 
 -module(bksw_wm_base).
 
 %% Complete webmachine callbacks
--export([init/1,
+-export([finish_request/2,
+         init/1,
          is_authorized/2,
-         finish_request/2,
+         malformed_request/2,
          service_available/2]).
 
 %% Helper functions
 -export([create_500_response/2]).
 
+-ifdef(TEST).
+-compile([export_all, nowarn_export_all]).
+-endif.
+
 -include("internal.hrl").
 
 %%
 %% Complete webmachine callbacks
 %%
 
+% why is this here?
+-include_lib("eunit/include/eunit.hrl").
+
+% number of seconds in 1 week
+-define(WEEK, 604800).
+
+-define(NOT_APPLICABLE, "1").
+
 init(Config) ->
     {ok, bksw_conf:get_context(Config)}.
 
+malformed_request(Req0, #context{auth_check_disabled=true} = Context) -> {false, Req0, Context};
+malformed_request(Req0, #context{                        } = Context) ->
+    HeadersRaw = mochiweb_headers:to_list(wrq:req_headers(Req0)),
+    Headers    = process_headers(HeadersRaw),
+    {RequestId, Req1} = bksw_req:with_amz_request_id(Req0),
+    try
+        case proplists:get_value('Authorization', HeadersRaw, undefined) of
```

Comment:
Should we be using wrq:get_req_header instead? It is case-insensitive, where proplists:get_value will be case-sensitive. 

`Headers` might also be used, because we do a case conversion there - but then it's all lowercase. 

Too, will the header field name be an atom `'Authorization'`? I would expect a string or binary.

Reply:
The spec called for a capitalized `'Authorization'` header to be used, but if you want it to be more lenient, or if that's a best practice, I'm happy to change it so by all means let me know.

The best way to answer the `'Authorization'` is-it-token-or-string/binary question would probably be for me to change  the code, run it through tests, and see if it breaks.  Having said that, since it is currently passing, I'm guessing it's correct?

Leaving this open for now until we resolve it one way or the other.  And I'll try to get answers to the token/string-binary question and report back.

Reply:
Given that's what the spec calls for, I'm ok keeping it requiring uppercase, but would like to make sure that it will always be an atom.

Reply:
I traced through the code.  `Headers` is the output of `process_headers/1`.  `process_headers/1` converts the keys of all KV pairs to lowercase strings, regardless of whether those keys are atoms or strings.  Therefore, using `Headers` here vs `HeadersRaw`, and grabbing "authorization," should kill all birds here with one stone:  1) atom or string is immaterial, 2) capitalization is immaterial.

Reply:
Done.